contributing.qmd

```diff
@@ -22,10 +22,251 @@ A file that has the name of your extension, the categories it falls into, and a
+to the `manifest.json`:
+
+```json {filename="manifest.json"} 
+{
+  ...
+  "extension": {
+    "name": "my-content-name",
+    "title": "My Content Name",
+    "description": "A lovely, detailed description of the content.",
+    "homepage": "https://github.com/posit-dev/connect-extensions/tree/main/extensions/my-content-name",
+    "version": "0.0.0"
+  }
+}
+```
+
+It is recommended to begin with `"version": "0.0.0"` to avoid triggering a
+release during development of your content. When you are ready to release to
+the Gallery check the [Adding content to the Connect Gallery](#adding-content-to-the-connect-gallery) section.
+
+### Language version constraints
+
+To ensure that content in the gallery is installable on as many Posit Connect
+servers as possible we use the language version constraints feature to broaden
+the language versions Connect can use for a piece of content.
+
+All content in the gallery should include `requires` specifications for the
+language(s) it utilizes.
+
+Here is an example of what needs to be included in a `manifest.json`:
+
+```json {filename="manifest.json"}
+{
+  ...
+  "environment": {
+    "python": {
+      "requires": ">=3.8, <4"
+    },
+    "r": {
+      "requires": ">=4.2, <5"
+    }
+  },
+  ...
+}
 ```
+
+## Adding content to the Connect Gallery
+
+Once your content has the requirements above it is ready to be added to the
+Connect Gallery.
+
+The Connect Gallery uses GitHub Workflows to automate releases of new content
+when pull requests in this repo merge into `main`.
+
+To add content to the Connect Gallery, follow the steps below:
+
+### Adding simple content
+
+"Simple content" refers to content that can be bundled into a TAR file without
+any additional steps. Most content will fall into this category.
+
+::: {.callout-note}
+If you can run
+```bash
+tar -czf my-extension-name.tar.gz ./extensions/my-extension-name
+```
+and the resulting TAR file can be published by uploading the bundle to Posit
+Connect, then it is piece of simple content.
+:::
+
+To add simple content look for the [`simple-extension-changes` section in the `.github/workflows/extensions.yml`](https://github.com/posit-dev/connect-extensions/blob/main/.github/workflows/extensions.yml#L31)
+file and add a new filter:
+
+```yml
+filters: |
+  ...
+  my-content-name: extensions/my-content-name/**
+```
+
+This will make any code changes to your content trigger a few things:
+
+- lint: ensuring your content has everything it needs to be added to the
+  gallery
+- package: creating a TAR file of the content's directory to test in pull
+  requests and to be released
+- release: When the `version` in the `manifest.json` is incrased and merged to
+  `main`, releases the content to users of Posit Connect
+
+Lastly set the `version` in the `manifest.json` file to your initial release.
+
+When the changes above are merged the first release will be kicked off, and
+your content will be in the gallery. 🚀
+
+### Adding custom built content
```

Comment:
The custom built content contains a lot of links to the github actions / workflows. I wonder if it would be easier to just link to the relevant examples, and rely on the comments rather than keeping these links updated + duplicating some of the information here.

Reply:
I like this idea of using our examples (well commented) as the "this is how to do this thing"